File: Quick Starts/ExceptionHandling/VB-UnityIntegration/ExceptionHandlingQuickStart.BusinessLayer/RecoverableUpdateException.vb
'===============================================================================
' Microsoft patterns & practices Enterprise Library
' Exception Handling Application Block QuickStart
'===============================================================================
' Copyright © Microsoft Corporation.  All rights reserved.
' THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY
' OF ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT
' LIMITED TO THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
' FITNESS FOR A PARTICULAR PURPOSE.
'===============================================================================


Public Class RecoverableUpdateException
    Inherits ApplicationException

    ''' <summary>
    '''  Default constructor
    ''' </summary>
    Public Sub New()
        MyBase.New()
    End Sub


    ''' <summary>
    ''' Initializes with a specified error message.
    ''' </summary>
    ''' <param name="message">A message that describes the error.</param>
    Public Sub New(ByVal message As String)
        MyBase.new(message)
    End Sub

    ''' <summary>
    ''' Initializes with a specified error 
    ''' message and a reference to the inner exception that is the cause of this exception.
    ''' </summary>
    ''' <param name="message">The error message that explains the reason for the exception.</param>
    ''' <param name="exception">The exception that is the cause of the current exception.</param>
    Public Sub New(ByVal message As String, ByVal exception As Exception)
        MyBase.New(message, exception)
    End Sub

    ''' <summary>
    ''' Initializes with serialized data.
    ''' </summary>
    ''' <param name="info">The object that holds the serialized object data.</param>
    ''' <param name="context">The contextual information about the source or destination.</param>
    Protected Sub New(ByVal info As SerializationInfo, ByVal context As StreamingContext)
        MyBase.new(info, context)
    End Sub
End Class

File: Quick Starts/ExceptionHandling/CS/ExceptionHandlingQuickStart.BusinessLayer/RecoverableUpdateException.cs
//===============================================================================
// Microsoft patterns & practices Enterprise Library
// Exception Handling Application Block QuickStart
//===============================================================================
// Copyright © Microsoft Corporation.  All rights reserved.
// THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY
// OF ANY KIND, EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT
// LIMITED TO THE IMPLIED WARRANTIES OF MERCHANTABILITY AND
// FITNESS FOR A PARTICULAR PURPOSE.
//===============================================================================

using System;
using System.Runtime.Serialization;

namespace ExceptionHandlingQuickStart.DataLayer
{
	/// <summary>
	/// 
	/// </summary>
	public class RecoverableUpdateException : ApplicationException 
	{
		/// <summary>
		/// Default constructor
		/// </summary>
		public RecoverableUpdateException() : base() 
		{
		}

		/// <summary>
		/// Initializes with a specified error message.
		/// </summary>
		/// <param name="message">A message that describes the error.</param>
		public RecoverableUpdateException(string message) : base(message) 
		{
		}

		/// <summary>
		/// Initializes with a specified error 
		/// message and a reference to the inner exception that is the cause of this exception.
		/// </summary>
		/// <param name="message">The error message that explains the reason for the exception.
		/// </param>
		/// <param name="exception">The exception that is the cause of the current exception. 
		/// If the innerException parameter is not a null reference, the current exception 
		/// is raised in a catch block that handles the inner exception.
		/// </param>
		public RecoverableUpdateException(string message, Exception exception) : 
			base(message, exception) 
		{
		}

		/// <summary>
		/// Initializes with serialized data.
		/// </summary>
		/// <param name="info">The object that holds the serialized object data.</param>
		/// <param name="context">The contextual information about the source or destination.
		/// </param>
		protected RecoverableUpdateException(SerializationInfo info, StreamingContext context) :
			base(info, context) 
		{
		}
	}
}
